pkg/cluster/spec/spec.go

```diff
@@ -376,6 +378,20 @@ func (s *Specification) GetPDList() []string {
 	return pdList
 }
 
+// AdjustByVersion modify the spec by cluster version.
+func (s *Specification) AdjustByVersion(clusterVersion string) {
+	// CDC does not support data dir for version below v4.0.13, and also v5.0.0-rc, set it to empty.
+	if semver.Compare(clusterVersion, "v4.0.13") == -1 || clusterVersion == "v5.0.0-rc" {
+		log.Warnf(color.RedString("[WARN] data_dir is only supported with TiCDC version v4.0.13 or later, "+
```

Comment:
It seems will report this warning once the version is lower than `v4.0.13`, no matter if the user configure the data directory.